src/net/gateway-archive.ts

```diff
@@ -0,0 +1,151 @@
+import { FileApi } from "../api/file";
+import { GatewayArchiveError } from "../errors/gateway-archive";
+import { TextRecordKeys } from "../models/entity"
+import { VochainProcessStatus } from "../models/protobuf";
+import { getEnsPublicResolverByNetwork } from "../util/ens";
+import { ContentUri } from "../wrappers/content-uri"
+import { EthNetworkID, IGatewayClient, IGatewayDVoteClient } from "../common";
+
+export interface IArchiveResponseBody {
+    process?: {
+        [k: string]: any
+    }
+    results?: {
+        [k: string]: any
+    },
+}
+
+export namespace GatewayArchive {
+
+    /**
+     * Fetch the full state of the given processId on the process archive
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    export function getProcessStateArchive(processId: string, gateway: IGatewayClient, errorMessage: string) {
+        return getProcess(processId, gateway, errorMessage)
+    }
+
+    /**
+     * Fetch the headers of the given processId on the archive.
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    export function getProcessSummaryArchive(processId: string, gateway: IGatewayClient, errorMessage: string) {
+        return getProcess(processId, gateway, errorMessage)
+            .then((response) => {
+                response.process.envelopeHeight = response.results.envelopeHeight
+                return {
+                    processSummary: response.process
+                }
+            })
+    }
+
+    /**
+     * Retrieves the archive cumulative weight that has been casted in votes for the given process ID.
+     *
+     * @param processId
+     * @param gateway
+     * @param errorMessage
+     */
+    export function getResultsWeightArchive(processId: string, gateway: IGatewayClient, errorMessage: string) {
+        return getProcess(processId, gateway, errorMessage)
+            .then((response) => {
+                return {
```

Comment:
Why don't we return the value right away, instead of wrapping it into an object `{weight}` ?
If the caller needs to adapt something to its particular usage... it's its responsibility, not ours